### DISCUSSION TITLE
dbaclone and docker

### DISCUSSION THREAD

**Discussion by CarrieAnne** [sqlcollaborative/dbaclone#124] [Q&A]
Are there any general guidelines or preferably step by step instructions on how to use dbaclone and docker?  I've tried to follow the online presentations but seem to be missing some steps, probably around mounting the VHDX correctly for the container, and giving correct permissions.  I'm using Windows 10, WSL2 and the mcr.microsoft.com/mssql/server:2019-latest image.  I'd be grateful for any pointers.

**Comment by sanderstad** [comment]
In my presentations I already had an image available because that takes time to do.

Because you’re dealing with Docker you need to make sure the SQL Server instance, which is the guest, has access to your local directory on the host.

For example, the map c:\SQLData\Docker\Clone should be shared in the docker file as a share.

The example script I used for the sessions can also be found on my GitHub page:

 https://github.com/sanderstad/Presentations/tree/master/Solving%20the%20Dev%20DB%20problem%20with%20Docker%20and%20dbaclone

 

From: Millstone1998 ***@***.***> 
Sent: Thursday, 24 June 2021 16:27
To: sqlcollaborative/dbaclone ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [sqlcollaborative/dbaclone] dbaclone and docker (#124)

 

Are there any general guidelines or preferably step by step instructions on how to use dbaclone and docker? I've tried to follow the online presentations but seem to be missing some steps, probably around mounting the VHDX correctly for the container, and giving correct permissions. I'm using Windows 10, WSL2 and the mcr.microsoft.com/mssql/server:2019-latest image. I'd be grateful for any pointers.

**Comment by CarrieAnne** [comment]
I've come back to looking at this again and am still unable to make it work with my setup.  I believe this is because the mounting of the vhdx inside the container isn't currently supported on Windows 10, and is in a Preview only [Mount a VHD in WSL](https://docs.microsoft.com/en-gb/windows/wsl/wsl2-mount-disk#mount-a-vhd-in-wsl).  Could it be that you are not using WSL2 or possibly using a Linux based laptop rather than Windows?

**Comment by sanderstad** [comment]
The way to get this working is by sharing the image and clone share for all users to be accessible.
Then in Linux you need to connect to the share using CIFS and mount it.

The clones will then be able to created without being mounted directly. After that you can attach the database.

This is kind of tedious but like you said there is no direct way to get this working because it's kinda not supported
